README.md

```diff
@@ -92,6 +92,38 @@ wss.close({code: 4000, reason: 'Game over'});
 ```
 
 
+## Mapping to the protocol
+
+There is a 1:1 mapping between WebSocket messages and stream chunks.
+
+Each call to `read()` returns one WebSocket message. If a message is split into
+multiple frames on the wire, it won't be returned by `read()` until the final
+chunk (the one with the FIN flag set) arrives.
+
+When `read()` is not called, the browser and operating system will still buffer
+data to some extent, so backpressure will not be detected immediately by the
+server.
+
+Text messages appear in JavaScript as strings. Binary messages appear as
+ArrayBuffer objects.
+
+A clean close will result in `read()` returning an object with `done` set to
+true. An unclean close will result in a rejected promise.
+
+Each call to `write()` (or chunk that is piped into the `writable`) will be
+converted to one message. BufferSource (ArrayBuffer or TypedArray) objects will
```

Comment:
The browser may split the message into multiple frames.

Reply:
Thanks!